test/hpr_http_roaming_packet_SUITE.erl

```diff
@@ -0,0 +1,1588 @@
+    PubKeyBin = libp2p_crypto:pubkey_to_bin(PubKey),
+
+    ok = start_uplink_listener(),
+
+    SendPacketFun = fun(DevAddr) ->
+        GatewayTime = erlang:system_time(millisecond),
+        PacketUp = test_utils:frame_packet_uplink(
+            ?UNCONFIRMED_UP,
+            PubKeyBin,
+            SigFun,
+            DevAddr,
+            0,
+            #{timestamp => GatewayTime}
+        ),
+        hpr_routing:handle_packet(PacketUp),
+        {ok, PacketUp, GatewayTime}
+    end,
+
+    uplink_test_route(),
+
+    {ok, PacketUp, GatewayTime} = SendPacketFun(?DEVADDR_ACTILITY),
+    Payload = hpr_packet_up:payload(PacketUp),
+    Region = hpr_packet_up:region(PacketUp),
+    PacketTime = hpr_packet_up:timestamp(PacketUp),
+
+    {
+        ok,
+        #{<<"TransactionID">> := TransactionID, <<"ULMetaData">> := #{<<"FNSULToken">> := Token}},
+        _Request,
+        {200, _RespBody}
+    } = http_rcv(
+        #{
+            <<"ProtocolVersion">> => <<"1.1">>,
+            <<"SenderNSID">> => fun erlang:is_binary/1,
+            <<"DedupWindowSize">> => fun erlang:is_integer/1,
+            <<"TransactionID">> => fun erlang:is_number/1,
+            <<"SenderID">> => <<"0xC00053">>,
+            <<"ReceiverID">> => ?NET_ID_ACTILITY_BIN,
+            <<"MessageType">> => <<"PRStartReq">>,
+            <<"PHYPayload">> => hpr_http_roaming_utils:binary_to_hexstring(Payload),
+            <<"ULMetaData">> => #{
+                <<"DevAddr">> => ?DEVADDR_ACTILITY_BIN,
+                <<"DataRate">> => hpr_lorawan:datarate_to_index(
+                    Region,
+                    hpr_packet_up:datarate(PacketUp)
+                ),
+                <<"ULFreq">> => hpr_packet_up:frequency_mhz(PacketUp),
+                <<"RFRegion">> => erlang:atom_to_binary(Region),
+                <<"RecvTime">> => hpr_http_roaming_utils:format_time(GatewayTime),
+                <<"FNSULToken">> => fun erlang:is_binary/1,
+                <<"GWCnt">> => 1,
+                <<"GWInfo">> => [
+                    #{
+                        <<"RFRegion">> => erlang:atom_to_binary(Region),
+                        <<"RSSI">> => hpr_packet_up:rssi(PacketUp),
+                        <<"SNR">> => hpr_packet_up:snr(PacketUp),
+                        <<"DLAllowed">> => true,
+                        <<"ID">> => hpr_http_roaming_utils:binary_to_hexstring(
+                            hpr_utils:pubkeybin_to_mac(PubKeyBin)
+                        )
+                    }
+                ]
+            }
+        }
+    ),
+
+    ?assertMatch(
+        {ok, TransactionID, 'US915', PacketTime, <<"127.0.0.1:3002/uplink">>, sync},
+        hpr_http_roaming:parse_uplink_token(Token)
+    ),
+
+    ok.
+
+http_class_c_downlink_test(_Config) ->
+    %%
+    %% Forwarder : HPR
+    %% Roamer    : partner-lns
+    %%
+    ok = start_forwarder_listener(),
+    ok = start_roamer_listener(#{callback_args => #{flow_type => async}}),
+
+    %% 1. N/A
```

Comment:
?

Reply:
Once migrated from Packet Purchaser, the first step of the test was not relevant.
I will re-number the comments.